shared/src/main/scala-3/magnolify/shims/package.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package magnolify
+
+import scala.util.hashing.MurmurHash3
+import scala.collection.compat.Factory
```

Comment:
is it really needed in scala3? I thought it is only 2.12 .vs 2.13 issue

Reply:
scala 2.13 and scala 3 have the same collection IIRC, but we still need the shim layer to get the code cross-build with scala 2.12. I think this can be removed when we stop 2.12 support.